File: PhotoRollPlayer/ViewController.m
//
//  ViewController.m
//  PhotoRollPlayer
//
//  Created by wangshiyu13 on 15/6/25.
//  Copyright © 2015年 manyi. All rights reserved.
//

#import "ViewController.h"
#import "PhotoRollPlayerView.h"

@interface ViewController ()
@property (nonatomic, strong) PhotoRollPlayerView *rollPlayerView;
@end

@implementation ViewController

- (PhotoRollPlayerView *)rollPlayerView {
    if (_rollPlayerView == nil) {
        _rollPlayerView = [[PhotoRollPlayerView alloc] initWithFrame:CGRectMake(0, 0, self.view.bounds.size.width, 200)];
        NSMutableArray *arrayM = [NSMutableArray array];
        for (int i = 1; i < 5; i++) {
            PhotoButton *button = [[PhotoButton alloc] initWithFrame:CGRectMake(0, 0, [UIScreen mainScreen].bounds.size.width, 200)];
            button.contentImageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:[NSString stringWithFormat:@"%d.jpg", i]]];
            [button addTarget:self action:@selector(click:) forControlEvents:UIControlEventTouchUpInside];
            [arrayM addObject:button];
        }
        _rollPlayerView.viewArray = arrayM;
    }
    return _rollPlayerView;
}

- (void)click:(PhotoButton *)btn {
    NSLog(@"%@", btn);
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.view addSubview:self.rollPlayerView];
}

@end

File: PhotoRollPlayer/ViewController.swift
//
//  ViewController.swift
//  
//
//  Created by 汪诗雨 on 15/7/28.
//
//

import UIKit

class ViewController: UIViewController {
    lazy var rollPlayerView: PhotoRollPlayerView = {
        let view: PhotoRollPlayerView = PhotoRollPlayerView(frame: CGRectMake(0, 0, self.view.bounds.size.width, 200))
        let array = NSMutableArray()
        for i in 1..<5 {
            let button: PhotoButton = PhotoButton(frame: CGRectMake(0, 0, self.view.bounds.size.width, 200))
            button.contentImageView = UIImageView(image: UIImage(named: "\(i).jpg"))
            button.addTarget(self, action: "click:", forControlEvents: UIControlEvents.TouchUpInside)
            array.addObject(button)
        }
        view.viewArray = array as [AnyObject]
        return view
        }()
    
    func click(sender: PhotoButton) {
        print("click--------------\(sender)")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.addSubview(self.rollPlayerView)
        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
